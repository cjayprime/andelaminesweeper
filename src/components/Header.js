import React, { Component } from 'react';
import {
    TouchableOpacity,
    View,
    Text,
    Image
} from 'react-native';

import Button from './Button';

import { flag, timer, reload, arrowDown } from '../assets/images';
import * as Style from '../assets/styles';

import { Constants } from '../utils';

export default class Container extends Component{
    state = {
        countdown: '00:00',
        seconds: 0
    };
    
    componentDidMount(){
        setInterval(this.time.set, 1000);
    };

    componentDidUpdate(prevProps){
        if(prevProps.clear !== this.props.clear){
            this.time.reset();
        }
    };

    time = {
        pad: (time) => {
            var timeString = time + "";
            if (timeString.length < 2) {
                return "0" + timeString;
            } else {
                return timeString;
            }
        },
        reset: () => {
            this.setState({
                countdown: '00:00',
                seconds: 0
            });
        },
        set: () => {
            let { seconds } = {...this.state};
            ++seconds;
            let countdown = this.time.pad(parseInt(seconds / 60));
            countdown += ':' + this.time.pad(seconds % 60);
            this.setState({
                countdown,
                seconds
            });
        }
    };
    
    refresh = () => {
        const { onChange } = this.props;
        onChange('Hard', true);
        this.time.reset();
    };

    render(){
        const height = Constants.HEADER_HEIGHT;
        const { difficulty, open, mines } = this.props;
        const { countdown } = this.state;
        return (
            <>
                <View style={[Style.header.container, { height }]}>
                    <Button label={difficulty} onPress={open} image={arrowDown}/>
                    <View style={Style.header.flag}>
                        <Image source={flag} style={Style.header.image} />
                        <Text style={Style.header.text}>{mines}</Text>
                    </View>
                    <View style={Style.header.timer}>
                        <Image source={timer} style={Style.header.image} />
                        <Text style={Style.header.text}>{countdown}</Text>
                    </View>
                    <TouchableOpacity onPress={this.refresh}  style={Style.header.reload}>
                        <Image source={reload} style={Style.header.image} />
                    </TouchableOpacity>
                </View>
            </>
        );
    }
}