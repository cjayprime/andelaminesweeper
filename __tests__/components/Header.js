import React from 'react';
import Header from '../../src/components/Header';

import renderer from 'react-test-renderer';

jest.useFakeTimers();
describe('Header Test', () => {
    test('Stability', () => {
        const tree = renderer.create(
            <Header
                open={() => {}}
                clear={false}
                difficulty={'Easy'}
                onChange={() => {}}
                mines={100}
                pause={false}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});