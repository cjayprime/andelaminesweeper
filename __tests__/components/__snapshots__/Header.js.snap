// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Test Stability 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "elevation": 5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
      },
      Object {
        "height": 44,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginTop": -10,
        "width": "30%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#000",
          "borderRadius": 5,
          "elevation": 3,
          "flexDirection": "row",
          "height": 30.799999999999997,
          "justifyContent": "space-around",
          "marginTop": 10,
          "opacity": 1,
          "width": "85%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 15,
          }
        }
      >
        Easy
      </Text>
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/images/arrowDown.png",
          }
        }
        style={
          Object {
            "height": 15,
            "width": 15,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "15%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/flag.png",
        }
      }
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 25,
        }
      }
    >
      100
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "35%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/timer.png",
        }
      }
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 25,
        }
      }
    >
      00:00
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "width": "10%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/reload.png",
        }
      }
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    />
  </View>
</View>
`;
